CMake/resolve_dependency_modules/cpr.cmake

```diff
@@ -20,6 +20,9 @@ set(VELOX_CPR_SOURCE_URL
     "https://github.com/libcpr/cpr/archive/refs/tags/${VELOX_CPR_VERSION}.tar.gz"
 )
 
+set(curl_SOURCE BUNDLED)
+resolve_dependency(curl)
```

Comment:
Yes, you can refer to the sentence "the first such call will control how that dependency will be made available" in this [link](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable).

Reply:
I can still see the curl building log in cpr building, could help double check?
https://app.circleci.com/pipelines/github/facebookincubator/velox/39899/workflows/20a0d56a-51b9-4832-8d9f-96e5685a33f2/jobs/271655/parallel-runs/0/steps/0-106?invite=true#step-106-40221_31

Reply:
This is just a log in the cmake of cpr, and it will always be printed out. I just removed `FetchContent_MakeAvailable` in curl.cmake, so the compilation process is triggered by cpr with the correct compilation parameters. The role of `curl.cmake` in Velox is simply to define the source URL of curl through `FetchContent_Declare`.

Reply:
Still using the built-in curl, but now we can specify the source URL and version of curl.